Content.Server/Explosion/EntitySystems/ExplosionSystem.Processing.cs

```diff
@@ -0,0 +1,606 @@
+    public const int NukeArea = 400;
+
+    /// <summary>
+    ///     Used to limit explosion processing time. See <see cref="MaxProcessingTime"/>.
+    /// </summary>
+    internal readonly Stopwatch Stopwatch = new();
+
+    /// <summary>
+    ///     How many tiles to explode before checking the stopwatch timer
+    /// </summary>
+    internal static int TileCheckIteration = 1;
+
+    /// <summary>
+    ///     Queue for delayed processing of explosions. If there is an explosion that covers more than <see
+    ///     cref="TilesPerTick"/> tiles, other explosions will actually be delayed slightly. Unless it's a station
+    ///     nuke, this delay should never really be noticeable.
+    /// </summary>
+    private Queue<Func<Explosion?>> _explosionQueue = new();
+
+    /// <summary>
+    ///     The explosion currently being processed.
+    /// </summary>
+    private Explosion? _activeExplosion;
+
+    /// <summary>
+    ///     While processing an explosion, the "progress" is sent to clients, so that the explosion fireball effect
+    ///     syncs up with the damage. When the tile iteration increments, an update needs to be sent to clients.
+    ///     This integer keeps track of the last value sent to clients.
+    /// </summary>
+    private int _previousTileIteration;
+
+    /// <summary>
+    ///     Process the explosion queue.
+    /// </summary>
+    public override void Update(float frameTime)
+    {
+        if (_activeExplosion == null && _explosionQueue.Count == 0)
+            // nothing to do
+            return;
+
+        Stopwatch.Restart();
+        var x = Stopwatch.Elapsed.TotalMilliseconds;
+
+        var availableTime = MaxProcessingTime;
+
+        var tilesRemaining = TilesPerTick;
+        while (tilesRemaining > 0 && MaxProcessingTime > Stopwatch.Elapsed.TotalMilliseconds)
+        {
+            // if there is no active explosion, get a new one to process
+            if (_activeExplosion == null)
+            {
+                // EXPLOSION TODO allow explosion spawning to be interrupted by time limit. In the meantime, ensure that
+                // there is at-least 1ms of time left before creating a new explosion
+                if (MathF.Max(MaxProcessingTime - 1, 0.1f)  < Stopwatch.Elapsed.TotalMilliseconds)
+                    break;
+
+                if (!_explosionQueue.TryDequeue(out var spawnNextExplosion))
+                    break;
+
+                _activeExplosion = spawnNextExplosion();
+
+                // explosion spawning can be null if something somewhere went wrong. (e.g., negative explosion
+                // intensity).
+                if (_activeExplosion == null)
+                    continue;
+
+                _explosionCounter++;
+                _previousTileIteration = 0;
+
+                // just a lil nap
+                if (SleepNodeSys)
+                {
+                    _nodeGroupSystem.Snoozing = true;
+                    // snooze grid-chunk regeneration?
```

Comment:
This would require an engine PR to add this feature make `SetTiles()` store chunks in a to-be-updated queue.
Doable, but for now the approach of not setting tiles until the very end of the explosion works well enough, I'll probably add this later on, so that explosions can once again break tiles as they expand.